strawberry/starlite/controller.py

```diff
@@ -0,0 +1,399 @@
+
+            if "application/json" in content_type:
+                try:
+                    data = await request.json()
+                except json.JSONDecodeError:
+                    actual_response = Response(
+                        "Unable to parse request body as JSON",
+                        status_code=HTTP_400_BAD_REQUEST,
+                        media_type=MediaType.TEXT,
+                    )
+                    return actual_response
+            elif content_type.startswith("multipart/form-data"):
+                multipart_data = await request.form()
+                operations: Dict[str, Any] = multipart_data.get("operations", "{}")
+                files_map: Dict[str, List[str]] = multipart_data.get("map", "{}")
+                data = replace_placeholders_with_files(
+                    operations, files_map, multipart_data
+                )
+            else:
+                actual_response = Response(
+                    "Unsupported Media Type",
+                    status_code=HTTP_415_UNSUPPORTED_MEDIA_TYPE,
+                    media_type=MediaType.TEXT,
+                )
+                return actual_response
+
+            return await self.execute_request(
+                request=request,
+                data=data,
+                context=context,
+                root_value=root_value,
+            )
+
+        @websocket(websocket_path)
+        async def websocket_endpoint(
+            self,
+            socket: WebSocket,
+            context: CustomContext,
+            root_value: Any,
+        ) -> None:
+            async def _get_context():
+                return context
+
+            async def _get_root_value():
+                return root_value
+
+            preferred_protocol = self.pick_preferred_protocol(socket)
+            if preferred_protocol == GRAPHQL_TRANSPORT_WS_PROTOCOL:
+                await self.graphql_transport_ws_handler_class(
+                    schema=self._schema,
+                    debug=self._debug,
+                    connection_init_wait_timeout=self._connection_init_wait_timeout,
+                    get_context=_get_context,
+                    get_root_value=_get_root_value,
+                    ws=socket,
+                ).handle()
+            elif preferred_protocol == GRAPHQL_WS_PROTOCOL:
+                await self.graphql_ws_handler_class(
+                    schema=self._schema,
+                    debug=self._debug,
+                    keep_alive=self._keep_alive,
+                    keep_alive_interval=self._keep_alive_interval,
+                    get_context=_get_context,
+                    get_root_value=_get_root_value,
+                    ws=socket,
+                ).handle()
+            else:
+                # Code 4406 is "Subprotocol not acceptable"
+                await socket.close(code=4406)
+
+        def pick_preferred_protocol(self, socket: WebSocket) -> Optional[str]:
```

Comment:
Static. I would suggest you reconsider placing utilities on the controller class .